File: python/agentic_commerce_protocol/models/delegate_payment_request.py
# this file is @generated
import typing as t

from .address import Address
from .allowance import Allowance
from .common import BaseModel
from .payment_method_card import PaymentMethodCard
from .risk_signal import RiskSignal


class DelegatePaymentRequest(BaseModel):
    payment_method: PaymentMethodCard
    allowance: Allowance
    billing_address: t.Optional[Address] = None
    risk_signals: t.List[RiskSignal]
    metadata: t.Dict[str, str]

File: kotlin/src/main/kotlin/com/svix/agentic_commerce_protocol/DelegatePaymentRequest.kt
// This file is @generated
package com.svix.agentic_commerce_protocol

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class DelegatePaymentRequest(
    @SerialName("payment_method") val paymentMethod: PaymentMethodCard,
    val allowance: Allowance,
    @SerialName("billing_address") val billingAddress: Address? = null,
    @SerialName("risk_signals") val riskSignals: List<RiskSignal>,
    val metadata: Map<String, String>,
)
